lib/3scale_toolbox/commands/copy_command/copy_service.rb

```diff
@@ -23,183 +38,263 @@ def self.command
         end
 
         def run
-          source      = fetch_required_option(:source)
-          destination = fetch_required_option(:destination)
-          system_name = fetch_required_option(:target_system_name)
+          @source      = fetch_required_option(:source)
+          @destination = fetch_required_option(:destination)
+          @system_name = fetch_required_option(:target_system_name)
+          @service_id = arguments[:service_id]
 
-          copy_service(arguments[:service_id], source, destination, system_name)
+          process
         end
 
-        def compare_hashes(first, second, keys)
-          keys.map{ |key| first.fetch(key) } == keys.map{ |key| second.fetch(key) }
-        end
+        private
 
-        def provider_key_from_url(url)
-          url[/\w*@/][0..-2]
+        def create_remotes
+          @source_remote = get_remote source
```

Comment:
Is this method "creating" remotes? It just gets them.

Also, I don't really like methods that just set instance variables.
And I personally use instance variables only in the initializer. Everything else has to be done through accessors.